src/storage.js

```diff
@@ -0,0 +1,81 @@
+"use strict";
+
+const fs = require("fs");
+const fsextra = require("fs-extra");
+const path = require("path");
+const crypto = require("crypto");
+const helper = require("./helper");
+
+class Storage {
+	constructor() {
+		this.references = new Map();
+
+		// Ensures that a directory is empty.
+		// Deletes directory contents if the directory is not empty.
+		// If the directory does not exist, it is created.
+		fsextra.emptyDirSync(helper.getStoragePath());
```

Comment:
I mean, it's very likely that it's going to be enough in 90% cases, but I'm wondering how much time will happen before someone opens a ticket with a blown up FS :/ Maybe a month from now, or maybe never, who knows.

Do you think this is good enough, or do we need to add a setting for allowed storage size?

When I think about it, the logs will not be filled with proxified URLs, so that's not so annoying. But the problem of filling up space is still there, and whenever `Storage` is used for file uploads, we have to nuke this strategy: it would be pretty bad to destroy all uploaded files at startup!
I know this is not done now, just thinking forward since in his PR description, @xPaw mentioned:

> ` Storage` was designed to be used for file uploads in the future ;)